test/TestUtilities/Test.Utility/TestServer/PortReserverOfMockServer.cs

```diff
@@ -40,25 +41,35 @@ public class PortReserverOfMockServer : IDisposable
         /// </summary>
         /// <param name="basePath">The base path for all request URL's.
         /// Can be either null (default) for "/" or any "/"-prefixed string (e.g.:  /{GUID}).</param>
-        /// <param name="basePort">The base port for all request URL's.</param>
-        public PortReserverOfMockServer(string basePath = null, int basePort = 50231)
+        /// <param name="basePort">The base port for all request URL's. Defaults to system chosen available port,
+        /// at or below `65535`.</param>
+        public PortReserverOfMockServer(string basePath = null, int? basePort = null)
         {
             if (!string.IsNullOrEmpty(basePath) && (!basePath.StartsWith("/") || basePath.EndsWith("/")))
             {
                 throw new ArgumentException($"If provided, argument \"{nameof(basePath)}\" must start with and must not end with a slash (/)");
             }
 
-            if (basePort <= 0)
+            if (basePort is null)
             {
-                throw new InvalidOperationException();
+                // Port 0 means find an available port on the system.
+                var tcpListener = new TcpListener(IPAddress.Loopback, 0);
```

Comment:
Named param for knowing port allocation.

```suggestion
                var tcpListener = new TcpListener(IPAddress.Loopback, port: 0);
```